v2/apiserver/internal/lib/queue/writer.go

```diff
@@ -23,8 +23,16 @@ type WriterFactory interface {
 type Writer interface {
 	// Write writes the provided message to a specific queue (or similar channel)
 	// known to the implementation.
-	Write(ctx context.Context, message string) error
+	Write(ctx context.Context, message string, opts *MessageOptions) error
 	// Close executes implementation-specific cleanup. Clients MUST invoke this
 	// function when they are done with the Writer.
 	Close(context.Context) error
 }
+
+// MessageOptions represent options for the message being written via
+// the Writer.
+type MessageOptions struct {
+	// Ephemeral indicates whether or not the message should be durable/persisted
+	// (false) or not durable/persisted (true)
+	Ephemeral bool
```

Comment:
How would you feel about changing this to `Durable` so that we don't have to translate !Ephemeral == Durable in our heads?

fwiw, I think Durable should default to false (like Ephemeral does now) i.e. This would be the exact opposite behavior of what we have now. You'd need to _ask_ for a message to be durable. That tracks well with the fact that durability isn't a default on Artemis' end either. (And in general, durability isn't a default for message-oriented middleware.)

Reply:
Ah, can do.  Went this approach to maintain defaults with what they were, but if we are good with flipping them, agreed, sticking to Durable would be best.  I'll do this.

Reply:
> Went this approach to maintain defaults with what they were

I figured as much. I don't feel married to the the way things were.